week-4/Homework/mandatory/3-water-bottle.js

```diff
@@ -22,18 +22,52 @@ let bottle = {
   volume: 0,
   fillUp: function() {
     // calling this function should pour your bottle full (volume = 100);
+    if(this.volume === 100){
+      console.log("Bottle already full.");
+    }
+    else if(this.volume === 0) {
+      this.volume = this.volume + 100;
+    }
+    else if (this.volume > 0 && this.volume < 100 ){
+      console.log("Bottle not empty");
+    }
   },
   pour: function () {
     // calling this function should increase your bottle volume by 10 unit;
+    if (this.volume === 100){
+      console.log("Bottle is already full water volume cannot go beyond.");
+    }
+    else if(this.volume >=0 && this.volume <= 90){
+      this.volume = this.volume + 10;
+
+    } else{
+      this.volume = this.volume + 10;
+      console.log(`Whoops!!! Looks like you've changed your bottle to a bigger one, it went beyond its maximum capacity up to ${this.volume} unit.`)
+    }
   },
   drink: function() {
     // calling this function should decrease your bottle volume by 10 unit;
+    if(this.volume === 0){
+      console.log("Bottle Empty!");
+    }
+    else if( this.volume < 10){
+      console.log("Not enough water");
+    }
+    else{
+      this.volume = this.volume - 10;
+    }
   },
   isFull: function () {
-    // this function should return true if your bottle is empty;
+    // this function should return true if your bottle is full;
+    if(this.volume === 100){

```

Comment:
In these `if` statements make sure you have a condition for what happens if `this.volume === 100` evaluates to `false` - at the moment if `this.volume` was a value other than `100` then the return value would be `undefined`. You could also just return the expression `this.volume === 100` anyway as this will evaluate to either `true` or `false`.